cmd/soroban-rpc/lib/preflight/src/fees.rs

```diff
@@ -54,7 +50,8 @@ pub(crate) fn compute_host_function_transaction_data_and_min_fee(
             &soroban_resources.footprint,
         )
         .context("cannot estimate maximum transaction size")?,
-        contract_events_size_bytes: soroban_resources.contract_events_size_bytes,
+        contract_events_size_bytes: calculate_event_size_bytes(events)
```

Comment:
Not related to the current failures, but you should calculate the size of the contract events only, not all the diagnostic events (i.e. you need to filter based on the event type in that function).

Reply:
Ah, good to know.